SDK Showcase/SDK Showcase/Views/Categories/CategoriesList.swift

```diff
@@ -0,0 +1,30 @@
+//  Copyright © 2025 Onewelcome Mobile Identity. All rights reserved.
+
+import Foundation
+import SwiftUI
+
+struct CategoriesRepository {}
+
+struct CategoriesList: View {
+    private var interactor: CategoriesInteractor {
+        @Injected var interactors: Interactors
+        return interactors.categoriesInteractor
+    }
+    @State private var categories = [Category]()
+    @Injected private var appState: AppState
+
+    var body: some View {
+        NavigationView {
+            List(categories) { category in
+                NavigationLink(destination: CategoryView(category: category)) {
+                    Text(category.name)
+                }
+            }
+            .environmentObject(appState.system)
+            .onAppear {
+                categories = interactor.loadCategories()
```

Comment:
Shouldn't be just sth like `interactor.loadCategories()` and as a result `appState` should change and be reflected as a list of categories?

Reply:
You are right in case of categories being loaded to DB (as I described it in the [document](https://onewelcome.atlassian.net/wiki/spaces/VMT/pages/320463929349/iOS+OMI+SDK+Showcase+former+Example+App)).

But... in this case we are loading it into memory, meaning the interactor is a function returning to the variable. The variable is stored in the view. We cannot store it in the interactor which the responsibility it just to interact with data (not storing it itself).

In case we decide to go to DB we will use the document's approach (your suggestion).
In case we use JSON we create a Storage class and voila - that will be the place.